src/composables/auth.ts

```diff
@@ -70,33 +76,44 @@ export const useAuth = createCustomScopedComposable(() => {
       }).then(() => {
         isAuthenticated.value = true;
       });
-    } catch (error) {
-      return Promise.reject(error);
+    } if (!IS_MOBILE_APP) {
+      return authenticateWithPassword(password!).then(({ encryptionKey: newEncryptionKey }) => {
+        setEncryptionKey(newEncryptionKey);
+        isAuthenticated.value = true;
+      });
     }
+    return Promise.resolve();
   }
 
-  function logout() {
+  /**
+   * Checks if the user should be kept logged in based on the current session.
+   */
+  async function shouldKeepExtensionLoggedIn() {
+    return !!(await getSessionEncryptionKey());
+  }
+
+  async function logout() {
+    setEncryptionKey(null);
     isAuthenticated.value = false;
   }
 
   async function openSecureLoginModal() {
```

Comment:
This function is not opening the modal but the mainly checks if user is logged in and authenticated. And if something is wrong it eventually do some kind of action. I suggest to rename this function to `checkUserAccess` / `checkAccess` / `checkUserAuth` (I prefer this one) / `checkAuth` or anything similar.

What do you think @CedrikNikita ?